client/webserver/api.go

```diff
@@ -424,24 +424,30 @@ func (s *WebServer) apiPostBond(w http.ResponseWriter, r *http.Request) {
 	}
 	defer zero(pass)
 
-	feeBuffer, err := s.bondsFeeBuffer(assetID) // could also put it in postBondForm, with some work on the frontend
-	if err != nil {
-		s.writeAPIError(w, err)
-		return
-	}
-
-	_, err = s.core.PostBond(&core.PostBondForm{
-		Addr:      post.Addr,
-		Cert:      []byte(post.Cert),
-		AppPass:   pass,
-		Bond:      post.Bond,
-		Asset:     &assetID,
-		LockTime:  post.LockTime,
-		FeeBuffer: feeBuffer,
+	bondForm := &core.PostBondForm{
+		Addr:     post.Addr,
+		Cert:     []byte(post.Cert),
+		AppPass:  pass,
+		Bond:     post.Bond,
+		Asset:    &assetID,
+		LockTime: post.LockTime,
 		// Options valid only when creating an account with bond:
 		MaintainTier: post.Maintain,
 		MaxBondedAmt: post.MaxBondedAmt,
-	})
+	}
+
+	if post.FeeBuffer != nil {
+		bondForm.FeeBuffer = *post.FeeBuffer
+	} else {
+		feeBuffer, err := s.bondsFeeBuffer(assetID) // or let wallet use internal feeBuffer?
+		if err != nil {
+			s.writeAPIError(w, err)
+			return
+		}
+		post.FeeBuffer = &feeBuffer
```

Comment:
I don't understand this line. Is it supposed to be `bondForm.FeeBuffer = feeBuffer`? Why are we doing this in **webserver** and not **core**?

Reply:
> Is it supposed to be bondForm.FeeBuffer = feeBuffer

Oh, yes.

> Why are we doing this in webserver and not core?

We've always fetched the feeBuffer here and I have no objection to moving it to `core` level.